packages/fxa-settings/src/pages/Signin/SigninRecoveryCode/container.tsx

```diff
@@ -0,0 +1,107 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { RouteComponentProps, useLocation } from '@reach/router';
+import { hardNavigateToContentServer } from 'fxa-react/lib/utils';
+import { MozServices } from '../../../lib/types';
+import SigninRecoveryCode from '.';
+import { Integration, useAuthClient } from '../../../models';
+import LoadingSpinner from 'fxa-react/components/LoadingSpinner';
+import { useMutation } from '@apollo/client';
+import { CONSUME_RECOVERY_CODE_MUTATION } from './gql';
+import { useCallback } from 'react';
+import { getStoredAccountInfo, handleGQLError } from '../utils';
+import { SigninLocationState } from '../interfaces';
+import { useFinishOAuthFlowHandler } from '../../../lib/oauth/hooks';
+import { useValidatedQueryParams } from '../../../lib/hooks/useValidate';
+import { SigninQueryParams } from '../../../models/pages/signin';
+import { ConsumeRecoveryCodeResponse, SubmitRecoveryCode } from './interfaces';
+import AppLayout from '../../../components/AppLayout';
+import CardHeader from '../../../components/CardHeader';
+
+export const viewName = 'signin-recovery-code';
+
+export type SigninRecoveryCodeContainerProps = {
+  integration: Integration;
+  serviceName: MozServices;
+};
+
+export const SigninRecoveryCodeContainer = ({
+  integration,
+  serviceName,
+}: SigninRecoveryCodeContainerProps & RouteComponentProps) => {
+  const authClient = useAuthClient();
+  const { finishOAuthFlowHandler, oAuthDataError } = useFinishOAuthFlowHandler(
+    authClient,
+    integration
+  );
+  // TODO: FXA-9177, likely use Apollo cache here instead of location state
+  const location = useLocation() as ReturnType<typeof useLocation> & {
+    state: SigninLocationState;
+  };
+  const { queryParamModel } = useValidatedQueryParams(SigninQueryParams);
+  const { redirectTo } = queryParamModel;
+
+  const signinLocationState =
+    location.state && Object.keys(location.state).length > 0
```

Comment:
Should this check be more explicit? Seems like we are explcitly trying to get at the sessionToken. What if location.state is set to an object, but maybe it was primed with something other than account info... Perhaps this isn't a valid concern.